_posts/2023-06-06-Client-cert-auth-without-roles.adoc

```diff
@@ -0,0 +1,252 @@
+HTTP authentication.In this example we will be using certificates 
+signed by an example CA. We will be making use of the 
+`Dynamic-certificates` repository.
+
+toc::[]
+
+== Certification Generation
+
+First, clone the ```elytron-examples``` repo locally:
+
+```
+
+git clone https://github.com/wildfly-security-incubator/elytron-examples
+
+cd elytron-examples
+
+```
+
+Next, let's generate some client and server certificates that will be used in this example to set up two-way SSL between
+
+a server and two clients:
+
+```
+cd dynamic-certificates
+
+mvn clean install exec:java -Dexec.mainClass="org.wildfly.security.examples.CertificateGenerationExample" -Dexec.args="CN=Bob.Smith.123456 CN=Alice.Smith.456789"
+```
+
+The above command generates the following keystores and truststores in the ```dynamic-certificates/target``` directory:
+
+```
+client1.keystore
+
+client1.truststore
+
+client2.keystore
+
+client2.truststore
+
+server.keystore
+
+server.truststore
+```
+
+*  ```client1.keystore``` contains a certificate with distinguished name: ```CN=Bob.Smith.123456```
+
+*  ```client2.keystore``` contains a certificate with distinguished name: ```CN=Alice.Smith.456789```
+
+* Both client certificates are issued by an example certificate authority with distinguished name: ```CN=Elytron CA, ST=Elytron, C=UK, EMAILADDRESS=elytron@wildfly.org, O=Root Certificate Authority```
+
+*  ```server.truststore``` contains this certificate authority's certificate
+
+Next, convert the client keystores into PKCS12 format and import them into your browser so you can pick which one to 
+present to the server later on:
+
+```
+keytool -importkeystore -srckeystore client1.keystore -srcstoretype jks -destkeystore client1.keystore.pkcs12 -deststoretype pkcs12 -srcstorepass keystorepass -deststorepass keystorepass
+
+keytool -importkeystore -srckeystore client2.keystore -srcstoretype jks -destkeystore client2.keystore.pkcs12 -deststoretype pkcs12 -srcstorepass keystorepass -deststorepass keystorepass
+
+```
+
+Finally, copy the server.keystore and server.truststore files to your WildFly server instance:
+
+```
+
+cp $PATH_TO_ELYTRON_EXAMPLES/dynamic-certificates/target/server.* $WILDFLY_HOME/standalone/configuration
+
+```
+
+== Server Configuration
+
+The following set of instructions are going to be used to configure the wildfly server in order to 
+establish the mutual TLS connection. We will be deploying a simple web application under the 
+`elytron-examples/client-cert-auth-without-roles/simple-webapp` folder.
+
+Please navigate to the elytron server home directory and enter the following command.
+
+```
+
+$SERVER_HOME/bin/jboss-cli.sh --connect 
+
+```
+
+This will connect you to the server, after which you may proceed to configuring the server to establish the connection. 
+
+First we need to create two keystores inside the elytron subsystem. This will tell elytron where to go inder to find the 
+keystores. The command for this is: 
```

Comment:
Just a minor thing, for brevity, "This will tell elytron where to go inder to find the keystores. The command for this is: " could be removed and the first sentence could just be:

First we need to create two keystores inside the elytron subsystem: